src/core/graph.go

```diff
@@ -80,18 +79,27 @@ func (graph *BuildGraph) TargetOrDie(label BuildLabel) *BuildTarget {
 	return target
 }
 
-// Package retrieves a package from the graph by name
-func (graph *BuildGraph) Package(name string) *Package {
+// PackageByLabel retrieves a package from the graph using the appropriate parts of the given label.
+// The Name entry is ignored.
+func (graph *BuildGraph) PackageByLabel(label BuildLabel) *Package {
+	return graph.Package(label.PackageName, label.Subrepo)
+}
+
+// Package retrieves a package from the graph by name & subrepo, or nil if it can't be found.
+func (graph *BuildGraph) Package(name, subrepo string) *Package {
 	graph.mutex.RLock()
 	defer graph.mutex.RUnlock()
-	return graph.packages[name]
+	return graph.packages[packageKey{Name: name, Subrepo: subrepo}]
 }
 
-// PackageOrDie retrieves a package by name, and dies if it can't be found.
-func (graph *BuildGraph) PackageOrDie(name string) *Package {
-	pkg := graph.Package(name)
+// PackageOrDie retrieves a package by label, and dies if it can't be found.
+func (graph *BuildGraph) PackageOrDie(label BuildLabel) *Package {
+	pkg := graph.PackageByLabel(label)
 	if pkg == nil {
-		log.Fatalf("Package %s doesn't exist in graph", name)
+		if label.Subrepo != "" {
+			log.Fatalf("Package @%s//%s doesn't exist in graph", label.Subrepo, label.PackageName)
```

Comment:
don't bother with the `if` and just one argument and use `label.String()` ?

Reply:
`label.String` isn't quite the same (I don't want to print the target name), but good idea, I can reuse the `packageKey` struct for the same and drop the if.